api/controllers/ApplicationTypeController.js

```diff
@@ -243,69 +248,84 @@ module.exports = {
+        );
         if (premiumOrDropoffPageSelected && !loggedIn) {
-            return res.redirect(sails.config.customURLs.userServiceURL+'/usercheck?next=premiumCheck');
+            return res.redirect(
+                sails.config.customURLs.userServiceURL +
+                    '/usercheck?next=premiumCheck'
+            );
         }
 
         /**
          * Check the last application reference to ensure a unique reference is assigned to each new application.
          */
         ApplicationReference.findOne()
             .then(function (data) {
-
-                const uniqueApplicationId = HelperService.generateNewApplicationId(data, selectedServiceType);
+                const uniqueApplicationId =
+                    HelperService.generateNewApplicationId(
+                        data,
+                        selectedServiceType
+                    );
 
                 /**
                  * Query db for generated applicationid.  If none returned, application id is indeed unique.
                  * set a dummy value for the service type, as this will get changed on Page 2 when the user has actually chosen a service type
                  */
-                return sequelize.query('SELECT unique_app_id FROM "Application" WHERE unique_app_id = \'' + uniqueApplicationId + '\';')
-                    .spread(function (result) {
+                return sequelize
+                    .query(
+                        'SELECT unique_app_id FROM "Application" WHERE unique_app_id = \'' +
+                            uniqueApplicationId +
+                            "';"
+                    )
+                    .then(function ([result]) {
                         let user_id;
                         // add this to overcome issue with users coming from user management site
                         // where they must register.  registration/login disabled for now.
-                        if (req.session && req.session.passport && req.session.passport.user) {
+                        if (
+                            req.session &&
+                            req.session.passport &&
+                            req.session.passport.user
+                        ) {
                             user_id = req.session.passport.user;
-                        }
-                        else {
+                        } else {
                             user_id = 0;
                         }
 
                         if (result.length !== 0) {
-                            sails.log.warn('ID already taken, redirecting back to start page');
+                            sails.log.warn(
+                                'ID already taken, redirecting back to start page'
+                            );
 
                             res.redirect('/start');
 
-                            throw "ID already taken";
-
-                        } else
-                        {
+                            throw 'ID already taken';
```

Comment:
Wouldn't wrapping this in an `Error` be more helpful when debugging in future or does it not make a difference in this instance?

Reply:
Good spot, yes this should be wrapped around an Error object 👍